### PR TITLE
Bump dependencies version

### PR DESCRIPTION
**Pull request by prashanthpai**
* Bump up golang version required to 1.9.x
  Centos 7 is 1.9.4, latest stable release is 1.9.7
* etcd: Bump up etcd version required to 3.3.9
  We specifically need fixes related to auto compaction of etcd keyspace.
  These fixes are in an etcd version >= 3.3.3. Centos7 is still on 3.2.22 though.
* Bump up version of dep as we were using a really old version

### CODE REVIEW

File: scripts/install-reqs.sh [hunk 1]
```diff
@@ -39,5 +39,27 @@ install_gometalinter() {
   rm -f "$GOBINDIR/$LINTER_TARBALL"
 }
 
+install_etcd() {
```
Comment (kshlm) [inline scripts/install-reqs.sh]: We'd removed this! When?

Also, @sseshasa will be bumping up etcd version in Gopkg.toml in #1068.

**Comment by kshlm** [comment]
retest this please

File: scripts/install-reqs.sh (hunk 1 above)
Comment (prashanthpai) [reply scripts/install-reqs.sh]: I"m not sure if it was present as elastic etcd was the default.

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**